Vester/Private/Template/VesterTemplate.Tests.ps1

```diff
@@ -51,65 +83,55 @@ ForEach ($Test in $TestFiles) {
             # Use continue to skip this test and go to the next loop iteration
             continue
         }
-    }
+    }  
 
-    Describe -Name "$Scope Configuration: $TestName" -Fixture {
-        # Pull in $Title/$Description/$Desired/$Type/$Actual/$Fix from the test file
-        . $Test
+    # Runs through each test file on the below objects in the current scope
+    foreach($Test in $Tests)
+    {
 
-        # Pump the brakes if the config value is $null
-        If ($Desired -eq $null) {
-            Write-Verbose "Due to null config value, skipping test $TestName"
-            # Use continue to skip this test and go to the next loop iteration
-            continue
-        } Else {
-            $Datacenter = Get-Datacenter -Name $cfg.scope.datacenter -Server $cfg.vcenter.vc
-            # Use $Scope (parent folder) to get the correct objects to test against
-            # If changing values here, update the "$Scope -notmatch" test above as well
-            $InventoryList = switch ($Scope) {
-                'vCenter'    {$cfg.vcenter.vc}
-                'Datacenter' {$Datacenter}
-                'Cluster'    {$Datacenter | Get-Cluster -Name $cfg.scope.cluster}
-                'DSCluster'  {$Datacenter | Get-DatastoreCluster -Name $cfg.scope.dscluster}
-                'Host'       {$Datacenter | Get-Cluster -Name $cfg.scope.cluster | Get-VMHost -Name $cfg.scope.host}
-                'VM'         {$Datacenter | Get-Cluster -Name $cfg.scope.cluster | Get-VM -Name $cfg.scope.vm}
-                'Network'    {$Datacenter | Get-VDSwitch -Name $cfg.scope.vds}
-            }
-        } #If Desired
+        # Loops through each object in the inventory list for the specific scope.
+        # It runs one test at a time against each $Object and moves onto the next test.
+        foreach($Object in $Inventory)
+        {
```

Comment:
92-95 should move below 96-108. This ensures that for each test, only one Describe block is created for all objects, and the `$Desired -eq $null` check happens once for each test (instead of once for each object inside each test).